stl/inc/format

```diff
@@ -652,20 +664,25 @@ public:
     }
 
     constexpr void _On_zero() {
-        _Specs._Alignment = _Align::_None;
-        _Specs._Fill[0]   = _CharT{'0'};
+        _Specs._Leading_zero = true;
+        if (_Specs._Alignment == _Align::_None) {
+            _Specs._Fill[0] = _CharT{'0'};
+        }
     }
 
-    constexpr void _On_width(unsigned int _Width) {
+    constexpr void _On_width(int _Width) {
         _Specs._Width = _Width;
     }
 
-    constexpr void _On_precision(unsigned int _Precision) {
+    constexpr void _On_precision(int _Precision) {
         _Specs._Precision = _Precision;
     }
 
     constexpr void _On_type(_CharT _Type) {
-        _Specs._Type = static_cast<_CharT>(_Type);
+        if (_Type < 0 || _Type > (numeric_limits<signed char>::max)()) {

```

Comment:
Hmmm - are we guaranteed that `_CharT` is an integral type here? Do I correctly recall that we only need to support `char` and `wchar_t` and not user-defined character types?

Reply:
I recommend deferring consideration of micro-optimizations that don't affect ABI. We can add a project card to investigate, or a `// performance note` for things which are highly likely to be improvable.

Reply:
to respond to casey, yes, CharT is only CharT or wchar_t, it could be a multi-byte character of some kind, but that would be an illegal type specifier and this branch would be taken resulting in throwing an error (this is why we're comparing against 127 and not 255 here).

Reply:
Quick investigation shows that our compilers optimize this properly, ignore me. I'm thinking of cases in algorithms where we know that the difference of two iterators is non-negative but the compiler does not.